backend/src/api/pools-parser.ts

```diff
@@ -0,0 +1,118 @@
+import {readFileSync} from 'fs';
+import { DB } from '../database';
+import logger from '../logger';
+
+interface Pool {
+  name: string,
+  link: string,
+  regexes: string[],
+  addresses: string[],
+}
+
+class PoolsParser {
+  /**
+   * Parse the pools.json file, consolidate the data and dump it into the database
+   */
+  public async migratePoolsJson() {
+    logger.info('Importing pools.json to the database');
+    let connection = await DB.pool.getConnection();
+
+    // Check if the pools table does not have data already, for now we do not support updating it
+    // but that will come in a later version
+    let [rows] = await connection.query<any>({ sql: 'SELECT count(id) as count from pools;', timeout: 120000 });
+    if (rows[0].count !== 0) {
+      logger.info('Pools table already contain data, updating it is not yet supported, skipping.');
+      connection.release();
+      return;
+    }
+
+    logger.info('Open ../frontend/cypress/fixtures/pools.json');
+    const fileContent: string = readFileSync('../frontend/cypress/fixtures/pools.json','utf8');
```

Comment:
Add `https://raw.githubusercontent.com/btccom/Blockchain-Known-Pools/master/pools.json` to the `EXTERNAL_ASSETS` config so it will be updated to the latest on startup. Then just read `./pools.json`

Reply:
that JSON file should be in the default configuration

Reply:
> Add `https://raw.githubusercontent.com/btccom/Blockchain-Known-Pools/master/pools.json` to the `EXTERNAL_ASSETS` config so it will be downloaded on startup. Then just read `./pools.json`

I think it's better to keep a copy in our repo and update from time to time for now. Downloading it automatically on everyone's Core node server is not be a good idea IMO.

Also, are `EXTERNAL_ASSETS` automatically re-downloaded if they change?

Reply:
you shouldn't be reading from that file anyway as it's intended to be used as a mock for the e2e tests

Reply:
Fixed! :)